### PR TITLE
Add rarity colors to player inventory

### LINKED ISSUE #1923: Feature Request: Rarity Colors on Player Sheets/Customisable Rarity Colors
In GitLab by @TomChristoffer on Dec 9, 2021, 10:02

Requesting rarity colors for items, feats, spells, etc... in player character sheets, and not just npc sheets,
as well as a system setting for customizing said colors with hex codes.

### PR DESCRIPTION
**Pull request by GravenImageRD**
Closes #1923

image

Also spotted a bug with mystified items showing the mystified name in parentheses instead of the original name, will open a separate PR with a fix for that.

### CODE REVIEW

**Comment by CarlosFdez** [comment]
If I recall, this is something that people explicitly didn't want to do for character, but I don't know too much past that.

**Comment by GravenImageRD** [comment]
Ah, I was just combing through the issues and saw this one. Should the issue just be closed as "won't do"?

**Comment by xdy** [comment]
It was closed last time, so I nicked the code and put it in the Workbench. *I* certainly want it. :)

**Comment by stwlam** [comment]
I'd be ok with this for identified items.

**Comment by GravenImageRD** [comment]
Could make it a player setting to enable, defaulting to disabled. That way folks that want it can turn it on and those that don't can leave it off.

**Comment by ebuckle** [comment]
I'm definitely a fan 👍

**Comment by stwlam** [comment]
> Could make it a player setting to enable, defaulting to disabled. That way folks that want it can turn it on and those that don't can leave it off.

No setting is necessary; just don't show it to players if the item isn't identified.

**Comment by GravenImageRD** [comment]
Then we should be all set, I just named that mystified item Mystified Item (No Rarity Color) to show that for mystified items they don't get the rarity color, so it should be working as requested, unless I'm missing something. :)

File: src/styles/actor/character/_inventory.scss [hunk 1]
```diff
@@ -3,6 +3,47 @@
...
+                &[data-item-rarity="rare"] {
+                    .item-name {
+                        h4 {
+                            border-color: $rarity-rare;
+                            color: $rarity-rare;
+                        }
+                    }
+                }
+
+                &[data-item-rarity="unique"] {
```
Comment (stwlam) [inline src/styles/actor/character/_inventory.scss:37]: Use a class instead of a data attribute

File: src/styles/actor/character/_inventory.scss [hunk 2]
```diff
@@ -5,42 +5,25 @@
...
                 }
 
-                &[data-item-rarity="unique"] {
-                    .item-name {
-                        h4 {
-                            border-color: $rarity-unique;
-                            color: $rarity-unique;
-                        }
-                    }
+                h4.item-rarity-unique {
```
Comment (stwlam) [inline src/styles/actor/character/_inventory.scss:24]:
```suggestion
                h4.rarity-unique {
```

File: src/styles/actor/character/_inventory.scss [hunk 3]
```diff
@@ -5,42 +5,25 @@
...
                 }
 
-                &[data-item-rarity="rare"] {
-                    .item-name {
-                        h4 {
-                            border-color: $rarity-rare;
-                            color: $rarity-rare;
-                        }
-                    }
+                h4.item-rarity-rare {
```
Comment (stwlam) [inline src/styles/actor/character/_inventory.scss:19]:
```suggestion
                h4.rarity-rare {
```

File: src/styles/actor/character/_inventory.scss [hunk 4]
```diff
@@ -5,42 +5,25 @@
...
                 }
 
-                &[data-item-rarity="uncommon"] {
-                    .item-name {
-                        h4 {
-                            border-color: $rarity-uncommon;
-                            color: $rarity-uncommon;
-                        }
-                    }
+                h4.item-rarity-uncommon {
```
Comment (stwlam) [inline src/styles/actor/character/_inventory.scss:14]:
```suggestion
                h4.rarity-uncommon {
```

File: src/styles/actor/character/_inventory.scss [hunk 5]
```diff
@@ -5,42 +5,25 @@
...
-            &[data-item-identified="true"] {
-                &[data-item-rarity="common"] {
-                    .item-name {
-                        h4 {
-                            border-color: $rarity-common;
-                            color: $rarity-common;
-                        }
-                    }
+            .item-name {
+                h4.item-rarity-common {
```
Comment (stwlam) [inline src/styles/actor/character/_inventory.scss:9]:
```suggestion
                h4.rarity-common {
```

File: static/templates/actors/partials/item-line.html [hunk 6]
```diff
@@ -1,8 +1,8 @@
 <li class="list-row">
-    <div class="item {{#if item.isTemporary}}temporary-item{{/if}}" data-item-id="{{item.id}}" data-item-is-container="{{isContainer}}" data-item-rarity="{{item.system.traits.rarity}}" data-item-identified="{{item.isIdentified}}">
+    <div class="item {{#if item.isTemporary}}temporary-item{{/if}}" data-item-id="{{item.id}}" data-item-is-container="{{isContainer}}">
         <div class="item-name rollable{{#if (and (ne @root.actor.type "loot") (not owner))}} long{{/if}}">
             <div class="item-image" style="background-image: url('{{item.img}}')"></div>
-            <h4>
+            <h4 {{#if (or user.isGM item.isIdentified)}}class="item-rarity-{{item.system.traits.rarity}}"{{/if}}>
```
Comment (stwlam) [inline static/templates/actors/partials/item-line.html:5]:
```suggestion
            <h4 {{#if (or user.isGM item.isIdentified)}}class="rarity-{{item.system.traits.rarity}}"{{/if}}>
```

**Pull request merged**